nixos/modules/services/misc/sonarr.nix

```diff
@@ -27,6 +41,31 @@ in
         '';
       };
 
+      settings = lib.mkOption {
+        inherit (settingsFormat) type;
+        description = "An attribute set containing Sonarr configuration settings.";
```

Comment:
I don't know where my previous comment went, but I think the option description should say that these settings will be re-applied on every service startup and effectively overwrite changes done in the web UI (silently).

One could argue that this is a breaking change and (1) it needs a changelog entry and (2) it should have an opt-in / opt-out mechanism. No strong opinion though, I'm fine with just updating `description`.

Reply:
Ah yeah completely zapped the backwards compatibility of `settings`.

We can make it `nullOr` and `null` by default, which would make it a no-op.

Reply:
hmm. I agree here, that having backwards-compatible is a good goal. But something like `UpdateMechanism` is something everybody would want. You can not update it from within the application.

But apart from that: i just discovered, that using env-vars is now possible: https://github.com/Sonarr/Sonarr/pull/6746

So i guess i have to rework this. specially for the API-KEY, ...

PS: sorry for the delayed response.

Reply:
That makes sense @ambroisie. Is a more secure option using [LoadCredential](https://systemd.io/CREDENTIALS/)?

Reply:
If possible, yes `LoadCredential` is _specifically_ made to load secrets.

Reply:
@ambroisie please have a look at my current idea.